### PR TITLE
[ruby] Create stable lambda names

### PR DESCRIPTION
**Pull request by AndreiDreyer**
* Create stable lambda names using surrounding structural infromation instead of simply having a global counter.

### CODE REVIEW

File: joern-cli/frontends/rubysrc2cpg/src/main/scala/io/joern/rubysrc2cpg/datastructures/RubyScope.scala [hunk 1]
```diff
@@ -191,6 +196,74 @@ class RubyScope(summary: RubyProgramSummary, projectRoot: Option[String])
...
+  def getNewClassTmp: String = {
+    val surroundingFullName = this.surroundingScopeFullName.getOrElse(RubyDefines.Main)
+
+    stack
+      .collectFirst {
+        case ScopeElement(scopeNode: NamespaceScope, _) => s"$surroundingFullName.${scopeNode.getNextClassTmp}"
+        case ScopeElement(node: TypeLikeScope, _)       => s"$surroundingFullName.${node.getNextClassTmp}"
+        case ScopeElement(node: MethodLikeScope, _)     => s"$surroundingFullName.${node.getNextClassTmp}"
+      }
+      .getOrElse(s"${surroundingFullName}.${this.getNextClassTmp}")
```
Comment (ml86) [inline joern-cli/frontends/rubysrc2cpg/src/main/scala/io/joern/rubysrc2cpg/datastructures/RubyScope.scala]: Why do we need this fallback handling here and for all the other kinds of temp variables?

File: joern-cli/frontends/rubysrc2cpg/src/test/scala/io/joern/rubysrc2cpg/querying/ClassTests.scala [hunk 2]
```diff
@@ -363,9 +363,9 @@ class ClassTests extends RubyCode2CpgFixture {
         |""".stripMargin)
 
     "generate a type decl with the associated members" in {
-      inside(cpg.typeDecl.nameExact("<anon-class-0>").l) {
+      inside(cpg.typeDecl.nameExact("Test0.rb:<main>.<anon-class-0>").l) {
```
Comment (ml86) [inline joern-cli/frontends/rubysrc2cpg/src/test/scala/io/joern/rubysrc2cpg/querying/ClassTests.scala]: In other tests you use `$Main` instead of `<main>` please do so here as well as in the rest of this file.

File: joern-cli/frontends/rubysrc2cpg/src/test/scala/io/joern/rubysrc2cpg/querying/MethodTests.scala [hunk 3]
```diff
@@ -1165,4 +1165,31 @@ class MethodTests extends RubyCode2CpgFixture {
       case xs => fail(s"Expected one call, got ${xs.code.mkString(",")}")
     }
   }
+
+  "tmp-vars in different methods should have same names" in {
```
Comment (ml86) [inline joern-cli/frontends/rubysrc2cpg/src/test/scala/io/joern/rubysrc2cpg/querying/MethodTests.scala]: Bad test description, "should both start from zero" is better.

**Review by ml86 — CHANGES_REQUESTED** [review]
I am also missing at least a single test where the tmp/anon counters are greater then zero.

**Review by ml86 — APPROVED** [review]
